packages/web/hooks/use-swap.ts

```diff
@@ -340,11 +331,11 @@ export function useSwapAssets({
   const { data: searchAssets, isLoading: isLoadingSearchAssets } =
     api.edge.assets.getAssets.useQuery(
```

Comment:
I suggest combining this query with the useInfiniteQuery to create a single query. By doing so, we would have less code, a simpler implementation, and we could take advantage of the benefits offered by paginated search

Reply:
Can we merge this and I address this in a future small PR?